cg/cli/backup.py

```diff
@@ -1,32 +1,110 @@
 """Backup related CLI commands."""
 import logging
 from pathlib import Path
-from typing import Iterable, List, Optional, Union
+from typing import Iterable, List, Optional, Tuple, Union
 
 import click
 import housekeeper.store.models as hk_models
 
 from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import FileExtensions
 from cg.constants.constants import DRY_RUN, FlowCellStatus
 from cg.constants.housekeeper_tags import SequencingFileTag
+from cg.exc import FlowCellError
 from cg.meta.backup.backup import BackupAPI, SpringBackupAPI
 from cg.meta.backup.pdc import PdcAPI
-from cg.meta.encryption.encryption import EncryptionAPI, SpringEncryptionAPI
+from cg.meta.encryption.encryption import (
+    EncryptionAPI,
+    FlowCellEncryptionAPI,
+    SpringEncryptionAPI,
+)
 from cg.meta.tar.tar import TarAPI
 from cg.models.cg_config import CGConfig
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
 from cg.store import Store
 from cg.store.models import Flowcell, Sample
 
 LOG = logging.getLogger(__name__)
 
 
+def _is_encryption_possible(
+    db_flow_cell: Flowcell,
+    flow_cell: FlowCellDirectoryData,
+    complete_file_path: Path,
+    pending_file_path: Path,
+) -> Tuple[bool, Union[str, None]]:
+    """Check if requirements for encryption are meet."""
+    if db_flow_cell and db_flow_cell.has_backup:
+        return False, f"Flow cell: {flow_cell.id} is already backed-up"
+    if not flow_cell.is_flow_cell_ready():
+        return False, f"Flow cell: {flow_cell.id} is not ready"
+    if complete_file_path.exists():
+        return False, f"Encryption already completed for flow cell: {flow_cell.id}"
+    if pending_file_path.exists():
+        return False, f"Encryption already started for flow cell: {flow_cell.id}"
+    return True, None
+
+
 @click.group()
 @click.pass_obj
 def backup(context: CGConfig):
     """Backup utilities"""
     pass
 
 
+@backup.command("encrypt-flow-cells")
+@DRY_RUN
+@click.pass_obj
+def encrypt_flow_cells(context: CGConfig, dry_run: bool):
+    """Encrypt flow cells."""
+    status_db: Store = context.status_db
+    flow_cell_encryption_api = FlowCellEncryptionAPI(
+        binary_path=context.encryption.binary_path, config=context.dict(), dry_run=dry_run
+    )
+    flow_cell_encryption_api.slurm_api.set_dry_run(dry_run=dry_run)
+    encrypt_dir: Path = Path(context.backup.encrypt_dir)
+    flow_cells_dir = Path(context.flow_cells_dir)
+    LOG.debug(f"Search for flow cells ready to encrypt in {flow_cells_dir}")
+    for flow_cell_dir in flow_cells_dir.iterdir():
+        if not flow_cell_dir.is_dir():
+            continue
+        LOG.debug(f"Found directory: {flow_cell_dir}")
+        try:
+            flow_cell = FlowCellDirectoryData(flow_cell_path=flow_cell_dir)
+        except FlowCellError:
```

Comment:
Why don't we have the `FlowCellDirectoryData` as an attribute in the `FlowCellEncryptionAPI` . We could Likely move all this logic that is performed in the CLI here to the API or not?

Reply:
Fixed